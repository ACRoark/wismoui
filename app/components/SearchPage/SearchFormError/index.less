@import '../../../styles/themes/index.less';

@search-form-error-cls-prefix: ~'@{dte-prefix}-search-form-error';

@desktop: ~'(min-width: 1025px)';

.@{search-form-error-cls-prefix} {
  align-items: flex-start;
  align-self: flex-start;
  background-color: @dte-red-light;
  color: @dte-red;
  display: flex;
  font-size: @dte-wismo-font-size-0;
  justify-content: space-between;
  margin-top: 30px;
  padding: 25px;

  @media @desktop {
    align-self: center;
    width: 230%;
  }

  .close-button {
    align-items: center;
    background-color: @dte-red-light;
    border: none;
    color: @dte-red;
    cursor: pointer;
    display: flex;
    height: 18px;
    justify-content: center;
    margin-top: 2px;
  }

  .icon {
    font-size: 1.875rem;
    font-weight: 600;
    margin-right: 20px;
  }

  .message {
    align-items: flex-start;
    display: flex;
  }
}
