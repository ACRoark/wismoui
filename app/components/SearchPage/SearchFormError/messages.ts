import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.search-form-error';

export default defineMessages({
  inactive: {
    id: `${scope}-inactive`,
  },
  invalid: {
    id: `${scope}-invalid`,
  },
  orderNotFound: {
    id: `${scope}-order-not-found`,
  },
  recaptchaFailed: {
    id: `${scope}-recaptcha-failed`,
  },
  unavailable: {
    id: `${scope}-unavailable`,
  },
  unexpected: {
    id: `${scope}-unexpected`,
  },
});
