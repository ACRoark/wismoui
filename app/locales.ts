export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
