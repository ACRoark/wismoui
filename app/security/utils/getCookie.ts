const getCookie = (name: string): string =>
  document.cookie.split('; ').reduce((previousValue: string, currentValue: string): string => {
    const parts = currentValue.split('=');

    return parts[0] === name ? decodeURIComponent(parts[1]) : previousValue;
  }, '');

export default getCookie;
