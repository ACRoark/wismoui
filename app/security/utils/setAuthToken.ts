import setCookie from './setCookie';

const setAuthToken = (value?: string): void => {
  if (value) {
    setCookie('token', `Bearer ${value}`, { days: 1, path: '/' });
  }
};

export default setAuthToken;
