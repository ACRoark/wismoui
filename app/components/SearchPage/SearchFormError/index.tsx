import React, { FC, ReactElement } from 'react';
import { FormattedMessage } from 'react-intl';

import useThunkDispatch from 'hooks/useThunkDispatch';
import { resetTracker } from 'store/actions';

import messages from './messages';

import './index.less';

interface ISearchFormError {
  error: string;
}

const SearchFormError: FC<ISearchFormError> = (props: ISearchFormError): ReactElement => {
  const { error } = props;

  const dispatch = useThunkDispatch();

  const handleClick = (): void => {
    dispatch(resetTracker());
  };

  const getMessage = (errorMsg: string): ReactElement => {
    switch (errorMsg) {
      case 'ORDER_INACTIVE':
        return <FormattedMessage {...messages.inactive} />;

      case 'ORDER_INVALID':
        return <FormattedMessage {...messages.invalid} />;

      case 'ORDER_NOT_FOUND':
        return <FormattedMessage {...messages.orderNotFound} />;

      case 'RECAPTCHA_FAILED':
        return <FormattedMessage {...messages.recaptchaFailed} />;

      case 'SERVER_UNAVAILABLE':
        return <FormattedMessage {...messages.unavailable} />;

      default:
        return <FormattedMessage {...messages.unexpected} />;
    }
  };

  return (
    <div className="dte-wismo-search-form-error">
      <div className="message">
        <span className="icon">!</span>
        {getMessage(error)}
      </div>
      <button className="close-button" onClick={handleClick} type="button">
        <svg width="18" height="18" viewBox="0 0 20 20" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
          <path
            d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
            stroke="currentColor"
            strokeWidth="2"
            strokeLinecap="round"
          />
        </svg>
      </button>
    </div>
  );
};

export default SearchFormError;
