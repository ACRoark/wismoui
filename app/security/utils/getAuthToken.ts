import getCookie from './getCookie';

const getAuthToken = (): string => getCookie('token');

export default getAuthToken;
