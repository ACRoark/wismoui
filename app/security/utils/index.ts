import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
