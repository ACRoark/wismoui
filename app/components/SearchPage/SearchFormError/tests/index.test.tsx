import React from 'react';
import { Provider } from 'react-redux';

import { SUPPORTED_LOCALES } from 'locales';

import { createMockStore, createSnapshotWithIntl  } from 'testing/utils';
import SearchFormError from '..';

const inactiveError = 'ORDER_INACTIVE';
const invalidError = 'ORDER_INVALID';
const orderNotFoundError = 'ORDER_NOT_FOUND';
const recaptchaError = 'RECAPTCHA_FAILED';
const unavailableError = 'SERVER_UNAVAILABLE';
const unexpectedError = 'well this was unexpected';

const store = createMockStore();

const testSearchFormErrorSnapshot = ( error: string, language: string): void => {
  const tree = createSnapshotWithIntl(<Provider store={store}><SearchFormError error={error} /></Provider>, language);

  expect(tree).toMatchSnapshot();
};

describe('<SearchFormError />', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    it(`should render correctly when a google reCAPTCHA error occurs and language is ${language}`, (): void => {
      testSearchFormErrorSnapshot(recaptchaError, language);
    });

    it(`should render correctly when an inactive order error occurs and language is ${language}`, (): void => {
      testSearchFormErrorSnapshot(inactiveError, language);
    });

    it(`should render correctly when an invalid order error occurs and language is ${language}`, (): void => {
      testSearchFormErrorSnapshot(invalidError, language);
    });

    it(`should render correctly when an order not found error occurs and language is ${language}`, (): void => {
      testSearchFormErrorSnapshot(orderNotFoundError, language);
    });

    it(`should render correctly when an unavailable server error occurs and language is ${language}`, (): void => {
      testSearchFormErrorSnapshot(unavailableError, language);
    });

    it(`should render correctly when an unexpected error occurs and language is ${language}`, (): void => {
      testSearchFormErrorSnapshot(unexpectedError, language);
    });
  });
});
