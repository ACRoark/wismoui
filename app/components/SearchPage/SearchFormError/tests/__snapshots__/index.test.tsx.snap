// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchFormError /> should render correctly when a google reCAPTCHA error occurs and language is en 1`] = `
<div
  className="dte-wismo-search-form-error"
>
  <div
    className="message"
  >
    <span
      className="icon"
    >
      !
    </span>
    We apologize for the inconvenience, but reCAPTCHA is currently unavailable. Please try again later.
  </div>
  <button
    className="close-button"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height="18"
      viewBox="0 0 20 20"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
        stroke="currentColor"
        strokeLinecap="round"
        strokeWidth="2"
      />
    </svg>
  </button>
</div>
`;

exports[`<SearchFormError /> should render correctly when a google reCAPTCHA error occurs and language is es 1`] = `
<div
  className="dte-wismo-search-form-error"
>
  <div
    className="message"
  >
    <span
      className="icon"
    >
      !
    </span>
    Disculpe las molestias, pero reCAPTCHA no está disponible actualmente. Por favor, inténtelo de nuevo más tarde.
  </div>
  <button
    className="close-button"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height="18"
      viewBox="0 0 20 20"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
        stroke="currentColor"
        strokeLinecap="round"
        strokeWidth="2"
      />
    </svg>
  </button>
</div>
`;

exports[`<SearchFormError /> should render correctly when an inactive order error occurs and language is en 1`] = `
<div
  className="dte-wismo-search-form-error"
>
  <div
    className="message"
  >
    <span
      className="icon"
    >
      !
    </span>
    The information provided is not associated with an active order. Please try again.
  </div>
  <button
    className="close-button"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height="18"
      viewBox="0 0 20 20"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
        stroke="currentColor"
        strokeLinecap="round"
        strokeWidth="2"
      />
    </svg>
  </button>
</div>
`;

exports[`<SearchFormError /> should render correctly when an inactive order error occurs and language is es 1`] = `
<div
  className="dte-wismo-search-form-error"
>
  <div
    className="message"
  >
    <span
      className="icon"
    >
      !
    </span>
    La información proporcionada no está asociada con un pedido activo. Inténtalo de nuevo.
  </div>
  <button
    className="close-button"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height="18"
      viewBox="0 0 20 20"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
        stroke="currentColor"
        strokeLinecap="round"
        strokeWidth="2"
      />
    </svg>
  </button>
</div>
`;

exports[`<SearchFormError /> should render correctly when an invalid order error occurs and language is en 1`] = `
<div
  className="dte-wismo-search-form-error"
>
  <div
    className="message"
  >
    <span
      className="icon"
    >
      !
    </span>
    The order number entered is not valid. Please try again.
  </div>
  <button
    className="close-button"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height="18"
      viewBox="0 0 20 20"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
        stroke="currentColor"
        strokeLinecap="round"
        strokeWidth="2"
      />
    </svg>
  </button>
</div>
`;

exports[`<SearchFormError /> should render correctly when an invalid order error occurs and language is es 1`] = `
<div
  className="dte-wismo-search-form-error"
>
  <div
    className="message"
  >
    <span
      className="icon"
    >
      !
    </span>
    El número de pedido ingresado no es válido. Inténtalo de nuevo.
  </div>
  <button
    className="close-button"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height="18"
      viewBox="0 0 20 20"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
        stroke="currentColor"
        strokeLinecap="round"
        strokeWidth="2"
      />
    </svg>
  </button>
</div>
`;

exports[`<SearchFormError /> should render correctly when an order not found error occurs and language is en 1`] = `
<div
  className="dte-wismo-search-form-error"
>
  <div
    className="message"
  >
    <span
      className="icon"
    >
      !
    </span>
    You have no active orders to track within our Order Tracker.
  </div>
  <button
    className="close-button"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height="18"
      viewBox="0 0 20 20"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
        stroke="currentColor"
        strokeLinecap="round"
        strokeWidth="2"
      />
    </svg>
  </button>
</div>
`;

exports[`<SearchFormError /> should render correctly when an order not found error occurs and language is es 1`] = `
<div
  className="dte-wismo-search-form-error"
>
  <div
    className="message"
  >
    <span
      className="icon"
    >
      !
    </span>
    No tiene órdenes activas para rastrear dentro de nuestro Rastreador de órdenes.
  </div>
  <button
    className="close-button"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height="18"
      viewBox="0 0 20 20"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
        stroke="currentColor"
        strokeLinecap="round"
        strokeWidth="2"
      />
    </svg>
  </button>
</div>
`;

exports[`<SearchFormError /> should render correctly when an unavailable server error occurs and language is en 1`] = `
<div
  className="dte-wismo-search-form-error"
>
  <div
    className="message"
  >
    <span
      className="icon"
    >
      !
    </span>
    Our system is currently unavailable. We apologize for the inconvenience. Please try again later.
  </div>
  <button
    className="close-button"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height="18"
      viewBox="0 0 20 20"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
        stroke="currentColor"
        strokeLinecap="round"
        strokeWidth="2"
      />
    </svg>
  </button>
</div>
`;

exports[`<SearchFormError /> should render correctly when an unavailable server error occurs and language is es 1`] = `
<div
  className="dte-wismo-search-form-error"
>
  <div
    className="message"
  >
    <span
      className="icon"
    >
      !
    </span>
    Nuestro sistema no está disponible actualmente. Disculpe las molestias. Por favor, inténtelo de nuevo más tarde.
  </div>
  <button
    className="close-button"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height="18"
      viewBox="0 0 20 20"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
        stroke="currentColor"
        strokeLinecap="round"
        strokeWidth="2"
      />
    </svg>
  </button>
</div>
`;

exports[`<SearchFormError /> should render correctly when an unexpected error occurs and language is en 1`] = `
<div
  className="dte-wismo-search-form-error"
>
  <div
    className="message"
  >
    <span
      className="icon"
    >
      !
    </span>
    An unexpected error has occured. We apologize for the inconvenience. Please try again later.
  </div>
  <button
    className="close-button"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height="18"
      viewBox="0 0 20 20"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
        stroke="currentColor"
        strokeLinecap="round"
        strokeWidth="2"
      />
    </svg>
  </button>
</div>
`;

exports[`<SearchFormError /> should render correctly when an unexpected error occurs and language is es 1`] = `
<div
  className="dte-wismo-search-form-error"
>
  <div
    className="message"
  >
    <span
      className="icon"
    >
      !
    </span>
    Ha ocurrido un error inesperado. Disculpe las molestias. Por favor, inténtelo de nuevo más tarde.
  </div>
  <button
    className="close-button"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height="18"
      viewBox="0 0 20 20"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
        stroke="currentColor"
        strokeLinecap="round"
        strokeWidth="2"
      />
    </svg>
  </button>
</div>
`;
